include/deal.II/base/aligned_vector.h

```diff
@@ -719,7 +720,7 @@ inline AlignedVector<T>::AlignedVector()
 
 template <class T>
 inline AlignedVector<T>::AlignedVector(const size_type size, const T &init)
-  : data_begin(nullptr)
+  : data_begin(nullptr, [](T *) { Assert(false, ExcInternalError()); })
```

Comment:
This doesn't seem right. Suppose someone calls this with `size = 0` and never resizes the vector. The object will then be in a state where its destructor unconditionally aborts.

A better solution would be to just have an empty function body for the deleter - it doesn't need to do anything.

Reply:
... also destructors should not throw (which might happen if `disable_abort_on_exception` is enabled).

Reply:
I should have checked the standard before writing that - it looks like unique_ptr doesn't call the deleter when it has a `nullptr`. The following code works:
```cpp
#include <iostream>
#include <memory>

template <typename T>
struct AlignedVector
{
  std::unique_ptr<T[], void (*)(T *)> data_begin;

    AlignedVector()
        : data_begin(nullptr, [](T *) { std::cout << "BBB\n"; std::abort();})
    {
        std::cout << "AAA\n";
    }
};

int main()
{
    AlignedVector<int> vec;
}
```